core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/validation/ContractValidationServiceImpl.java

```diff
@@ -134,26 +134,30 @@ public Result<ContractOffer> validateInitialOffer(ClaimToken token, ContractOffe
     }
 
     @Override
-    public boolean validateAgreement(ClaimToken token, ContractAgreement agreement) {
+    public Result<ContractAgreement> validateAgreement(ClaimToken token, ContractAgreement agreement) {
         var contractId = ContractId.parse(agreement.getId());
         if (!contractId.isValid()) {
-            return false;
+            return Result.failure(format("The contract id %s does not follow the expected scheme", agreement.getId()));
         }
 
         if (!isStarted(agreement) || isExpired(agreement)) {
-            return false;
+            return Result.failure("The agreement has not started yet or it has expired");
         }
 
         var agent = agentService.createFor(token);
         var contractDefinition = contractDefinitionService.definitionFor(agent, contractId.definitionPart());
         if (contractDefinition == null) {
-            return false;
+            return Result.failure("The ContractDefinition with id %s either does not exist or the access to it is not granted.");
```

Comment:
```suggestion
            return Result.failure(format("The ContractDefinition with id %s either does not exist or the access to it is not granted.", agreement.getId());
```
The value for "%s" is missing.